domain/src/test/java/org/oppia/domain/classify/rules/dragAndDropSortInput/DragDropSortInputIsEqualToOrderingClassifierProviderTest.kt

```diff
@@ -0,0 +1,132 @@
+import org.oppia.app.model.StringList
+import org.oppia.domain.classify.RuleClassifier
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [DragDropSortInputIsEqualToOrderingClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class DragDropSortInputIsEqualToOrderingClassifierProviderTest {
+  private val LIST_OF_SETS_OF_HTML_STRING_VALUE_1 = createListOfSetsOfHtmlStrings("this","is","a","test")
+  private val LIST_OF_SETS_OF_HTML_STRING_VALUE_2 = createListOfSetsOfHtmlStrings("is","this","a","test")
+  private val LIST_OF_SETS_OF_HTML_STRING_VALUE_3 = createListOfSetsOfHtmlStrings("this","is","a","failed","test")
+
+  @Inject
+  internal lateinit var dragDropSortInputIsEqualToOrderingClassifierProvider: DragDropSortInputIsEqualToOrderingClassifierProvider
+
+  private val isEqualToOrderingClassifierProvider:RuleClassifier by lazy {
+    dragDropSortInputIsEqualToOrderingClassifierProvider.createRuleClassifier()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testAnswer_testLisOfSetsOfHtmlString_sameValue_bothValuesMatch() {
+    val inputs = mapOf( "x" to LIST_OF_SETS_OF_HTML_STRING_VALUE_1)
+
+    val matches =
+      isEqualToOrderingClassifierProvider.matches(answer = LIST_OF_SETS_OF_HTML_STRING_VALUE_1, inputs = inputs)
+
+    Truth.assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testAnswer_testLisOfSetsOfHtmlString_differentValue_bothValuesDoNotMatch() {
+    val inputs = mapOf( "x" to LIST_OF_SETS_OF_HTML_STRING_VALUE_2)
+
+    val matches =
+      isEqualToOrderingClassifierProvider.matches(answer = LIST_OF_SETS_OF_HTML_STRING_VALUE_1, inputs = inputs)
+
+    Truth.assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer_testLisOfSetsOfHtmlString_differentLength_bothValuesDoNotMatch() {
+    val inputs = mapOf( "x" to LIST_OF_SETS_OF_HTML_STRING_VALUE_3)
+
+    val matches =
+      isEqualToOrderingClassifierProvider.matches(answer = LIST_OF_SETS_OF_HTML_STRING_VALUE_1, inputs = inputs)
+
+    Truth.assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testAnswer_testLisOfSetsOfHtmlString_incorrectInputMap_throwsException() {
+    val inputs = mapOf("y" to LIST_OF_SETS_OF_HTML_STRING_VALUE_1)
+
+    val exception = assertThrows(IllegalStateException::class) {
+      isEqualToOrderingClassifierProvider.matches(answer = LIST_OF_SETS_OF_HTML_STRING_VALUE_1, inputs = inputs)
+    }
+
+    Truth.assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'x' but had: [y]")
+  }
+
+
+  private fun createListOfSetsOfHtmlStrings(vararg items: String): InteractionObject {
+    val listOfSetsOfHtmlStrings = ListOfSetsOfHtmlStrings.newBuilder()
+      .addAllSetOfHtmlStrings(items.map { createHtmlStringList(it.split("")) })
```

Comment:
@BenHenning is that okay if I do it like this?

Reply:
I find it a bit hard to understand what the actual structure will look like for these answers. Can we be more explicit when setting them up? E.g. could we pass in a list of lists and assemble it that way?